src/psyclone/dynamo0p3.py

```diff
@@ -3178,36 +3202,42 @@ def gen_code(self, parent):
         self.cma_ops.initialise_cma_ops(invoke_sub)
 
         var_list = []
-        # loop over all unique function spaces used by the kernels in this
-        # invoke
+        # loop over all unique function spaces used by the kernels in
+        # this invoke
         for function_space in self.unique_fss():
             # Initialise information associated with this function space
-            invoke_sub.add(CommentGen(invoke_sub, ""))
-            invoke_sub.add(
-                CommentGen(invoke_sub, " Initialise number of DoFs for " +
-                           function_space.mangled_name))
-            invoke_sub.add(CommentGen(invoke_sub, ""))
+            if not dofs_only or not Config.get().distributed_memory:
```

Comment:
The two cases are different. This one needs to allow for the fact that we might need ndf and/or undf while the second one is only considering whether we need undf. undf is required if `DM==False` as it is the upper bound for the loop over dofs. However, if we're doing DM then we get the loop upper bound in a different way and do not need undf.

Reply:
I've updated the comment to try to make this clearer.

Reply:
As discussed, the logic is the same so if the cases are different then the logic needs looking at.

Reply:
Oh yes, having thought about it over lunch I can see that both cases boil down to "do we need undf or not?". I've changed them so that they're the same.

Reply:
Great.